library/gcm.c

```diff
@@ -278,7 +278,8 @@ int mbedtls_gcm_starts( mbedtls_gcm_context *ctx,
     else
     {
         memset( work_buf, 0x00, 16 );
-        MBEDTLS_PUT_UINT32_BE( iv_len * 8, work_buf, 12 );
+        MBEDTLS_PUT_UINT32_BE( iv_len >> 29, work_buf, 8  );
+        MBEDTLS_PUT_UINT32_BE( iv_len << 3,  work_buf, 12 );
```

Comment:
2 small nits:
- We have a macro: `MBEDTLS_PUT_UINT64_BE()` - please use this instead as it is neater than using 2 calls to `MBEDTLS_PUT_UINT32_BE()`
- `* 8` is more readable than `<< 3`, as the intent of the code is clearer (and most compilers will replace the multiply with a shift anyway).